packages/kube-object/src/kube-object.ts

```diff
@@ -118,13 +118,6 @@ export class KubeObject<
       );
     }
 
-    if (!isString(data.metadata.resourceVersion)) {
-      throw new KubeCreationError(
-        `Cannot create a KubeObject from an object without metadata.resourceVersion being a string`,
-        data,
-      );
-    }
-
     if (!isString(data.metadata.selfLink)) {
```

Comment:
I do not believe this kind of validation belongs in a constructor, if truly needed I believe is ought to be in a sanitisation feature